lib/arborist/build-ideal-tree.js

```diff
@@ -1206,11 +1206,17 @@ This is a one-time fix-up, please be patient...
           // the target is root, then we can't safely replace.
           for (const edge of p.edgesIn) {
             if (edge.peer) {
-              // root deps take precedence always.
-              // other peer deps on this node are irrelevant though.
-              if (edge.from.isRoot)
+              if (
+                // root deps take precedence always.
```

Comment:
Maybe this should just be `isTop` instead of `isRoot`? We probably don't need this check for link targets that aren't top nodes.

Reply:
oh nice top arborist surgeon eye there! it does work all the same 😊 so going with that instead!